File: 338-counting-bits/counting-bits.java
class Solution {
    public int[] countBits(int n) {
        int[] res = new int[n + 1];
        res[0] = 0;

        for (int i = 1; i <= n; i++) {
            res[i] = res[i >> 1] + (i & 1);
        }
        return res;
    }
}

File: 338-counting-bits/counting-bits.cpp
using namespace std;
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> arr(n + 1);
        for (int i = 0; i <= n; i++) {
            int cur = i, count = 0;
            while (cur != 0) {
                cur = cur & (cur - 1);
                count++;
            }
            arr[i] = count;
        }
        return arr;
    }
};
